src/main/java/com/gtnewhorizons/angelica/api/QuadProvider.java

```diff
@@ -3,13 +3,29 @@
 import com.gtnewhorizons.angelica.compat.mojang.BlockPos;
 import com.gtnewhorizons.angelica.compat.nd.Quad;
 import com.gtnewhorizons.angelica.utils.ObjectPooler;
-import java.util.List;
-import java.util.Random;
 import net.minecraft.block.Block;
 import net.minecraft.world.IBlockAccess;
 import net.minecraftforge.common.util.ForgeDirection;
 
+import java.util.List;
+import java.util.Random;
+
 public interface QuadProvider {
 
-    List<Quad> getQuads(IBlockAccess world, BlockPos pos, Block block, int meta, ForgeDirection dir, Random random, ObjectPooler<Quad> quadPool);
+    int R_MASK = 0xFF << 16;
+    int G_MASK = 0xFF << 8;
+    int B_MASK = 0xFF;
+
+    /**
+     * Returns the color packed as ABGR. If you want to vary colors within a block, just ignore the color passed into
+     * {@link #getQuads} and use whatever you want. You might want to override this to return a constant if you're
+     * extending a block with an expensive color call, like leaves.
+     */
+    default int getColor(IBlockAccess world, BlockPos pos, Block block, int meta, Random random) {
+
+        final int cin = block.colorMultiplier(world, pos.x, pos.y, pos.z);
+        return (0xFF << 24) | ((cin & B_MASK) << 16) | (cin & G_MASK) | ((cin & R_MASK) >>> 16);
+    }
+
+    List<Quad> getQuads(IBlockAccess world, BlockPos pos, Block block, int meta, ForgeDirection dir, Random random, int color, ObjectPooler<Quad> quadPool);
```

Comment:
Seems weird to rely on ObjectPooler which isn't in the API....  Perhaps make an interface as part of the API?

Same with Quad & BlockPos.

Reply:
ObjectPooler is required so the QuadProvider can pull quads from the provided pool from BlockRenderer or another API consumer, and List<Quad> is needed because that's what BlockRenderer consumes. BlockPos is there because we have it and I don't want to inflate the signature.

Should I move all three to the API package?

Reply:
Just make an interface that implements what's required and make the implementations implement the interface

IQuadProvider, and IQuad, IBlockPos or whatever appropriate name you want

Reply:
I thought we weren't using Hungarian notation in the API 🤔 (see `ThreadSafeISBRHFactory`)

Reply:
It's a weird mix of whatever we feel like at the time ;)